File: Python3.x/55-Jump_Game.py
# 贪心
class Solution:
    def canJump(self, nums: 'List[int]') -> 'bool':
        farest = nums[0]
        for i in range(len(nums)):
            if i <= farest and nums[i] + i > farest:
                farest = nums[i] + i
            if farest >= len(nums) - 1: return True
        return False

File: Algorithms/C++/55-Jump_Game.cpp
// Runtime: 12 ms, faster than 72.98% of C++ online submissions for Jump Game.
// Memory Usage: 9.8 MB, less than 99.48% of C++ online submissions for Jump Game.
class Solution {
public:
    bool canJump(vector<int>& nums) {
        int max_ = 0;
        for (int i = 0; i <= max_; ++i) {
            max_ = max(max_, i + nums[i]);
            if (max_ >= nums.size() - 1)
                return true;
        }
        return false;
    }
};
